Adds documentation for configuring registry for Nutanix (#63272)
openshift/openshift-docs: modules/installation-registry-storage-block-recreate-rollout-nutanix.adoc

// Module included in the following assemblies:
//
// * installing/installing_bare_metal/installing-bare-metal.adoc
// * installing/installing_baremetal/installing-bare-metal-network-customizations.adoc
// * installing/installing_baremetal/installing-restricted-networks-bare-metal.adoc
// * installing/installing_platform_agnostic/installing-platform-agnostic.adoc
// * registry/configuring_registry_storage/configuring-registry-storage-baremetal.adoc

:_content-type: PROCEDURE
[id="installation-registry-storage-block-recreate-rollout-nutanix_{context}"]
= Configuring block registry storage for Nutanix

> To allow the image registry to use block storage types such as Nutanix during upgrades as a cluster administrator, you can use the `Recreate` rollout strategy.

Reviewer: s/such as Nutanix/such as Nutanix Volumes

Seem it should be nutanix volumes

[IMPORTANT]
====
Block storage volumes, or block persistent volumes, are supported but not recommended for use with the image
registry on production clusters. An installation where the registry is
configured on block storage is not highly available because the registry cannot
have more than one replica.

If you choose to use a block storage volume with the image registry, you must use a filesystem Persistent Volume Claim (PVC).
====

.Procedure

. To set the image registry storage as a block storage type, patch the registry so that it uses the `Recreate` rollout strategy and runs with only one (`1`) replica:
+
[source,terminal]
----
$ oc patch config.imageregistry.operator.openshift.io/cluster --type=merge -p '{"spec":{"rolloutStrategy":"Recreate","replicas":1}}'
----

> 

Reviewer 2: Remove this extra line

.. Create a `pvc.yaml` file with the following contents to define a Nutanix `PersistentVolumeClaim` object:
+
[source,yaml]
----
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: image-registry-storage <1>
  namespace: openshift-image-registry <2>
spec:
  accessModes:
  - ReadWriteOnce <3>
  resources:
    requests:
      storage: 100Gi <4>
----
<1> A unique name that represents the `PersistentVolumeClaim` object.
<2> The namespace for the `PersistentVolumeClaim` object, which is `openshift-image-registry`.
<3> The access mode of the persistent volume claim. With `ReadWriteOnce`, the volume can be mounted with read and write permissions by a single node.
<4> The size of the persistent volume claim.

.. Create the `PersistentVolumeClaim` object from the file:
+
[source,terminal]
----
$ oc create -f pvc.yaml -n openshift-image-registry
----

. Edit the registry configuration so that it references the correct PVC:
+
[source,terminal]
----
$ oc edit config.imageregistry.operator.openshift.io -o yaml
----
+
.Example output
[source,yaml]
----
storage:
  pvc:
    claim: <1>
----

> <1> Creating a custom PVC allows you to leave the `claim` field blank for the default automatic creation of an `image-registry-storage` PVC.

Reviewer 2: ```suggestion
<1> By creating a custom PVC, you can leave the `claim` field blank for the default automatic creation of an `image-registry-storage` PVC.
```
The suggested revision avoids the potential for a dangling modifier and also avoids [anthropomorphic language](https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism) ("allows you").
